pubtools/_quay/tag_docker.py

```diff
@@ -645,6 +646,24 @@ def merge_manifest_lists_sign_images(
         if push_item.claims_signing_key:
             to_sign_entries = []
             to_sign_entries_internal = []
+
+            # for cosign sign also manifest list
+            to_sign_entries_internal.append(
+                SignEntry(
+                    repo=list(push_item.repos.keys())[0],
+                    reference="quay.io/"
+                    + self.target_settings["quay_namespace"]
+                    + "/"
+                    + internal_repo
+                    + ":"
+                    + tag,
+                    digest="sha256:"
+                    + hashlib.sha256(json.dumps(new_manifest_list).encode("utf-8")).hexdigest(),
```

Comment:
new_manifest_list may not be the uploaded manifest. It could also be raw_src_manifest:
https://github.com/release-engineering/pubtools-quay/blob/cosign-sign-ml/pubtools/_quay/tag_docker.py#L735

Reply:
updated